### PR TITLE
Os 186: Search support for numeric operators

### PR DESCRIPTION
**Pull request by pritha-tarento**
Search Api supports for numeric operators added.
Numeric operators extended supports are: **gte, gt, lte, lt**
NOTE: default is eq (=)
Below request payload change:
```
{
  "request": {
    "Teacher": {
      "serialNum": { "gt": 1 },
      "gender": { "eq": "GenderTypeCode-MALE" }
    }
  }
}
```

Numeric operator for range ( between )
```
{
  "request": {
    "Teacher": {
      "serialNum": {"between":[1,3]}
    }
  }
}
```
Tested: 
For ranges: if array [] is empty OR if array[1] contains single element OR  if array[2,1] descending order; Empty result returned.
if array elements size not 2 ; Exception raised

### CODE REVIEW

File: java/registry/src/main/java/io/opensaber/registry/dao/SearchDaoImpl.java [hunk 1]
```diff
@@ -29,21 +29,42 @@ public JsonNode search(Graph graphFromStore, SearchQuery searchQuery) {
...
+                    resultGraphTraversal = resultGraphTraversal.has(property, P.gt(genericValue));
+                    break;
+                case lt:
+                    resultGraphTraversal = resultGraphTraversal.has(property, P.lt(genericValue));
+                    break;
+                case gte:
+                    resultGraphTraversal = resultGraphTraversal.has(property, P.gte(genericValue));
+                    break;
+                case lte:
+                    resultGraphTraversal = resultGraphTraversal.has(property, P.gte(genericValue));
```
Comment (Hari-stackroute) [inline java/registry/src/main/java/io/opensaber/registry/dao/SearchDaoImpl.java]: I think it should be P.lte here

File: java/pojos/src/main/java/io/opensaber/pojos/Filter.java [hunk 2]
```diff
@@ -47,7 +49,7 @@ public void setPath(String path) {
 		this.path = path;
 	}
 
-	public String getOperator() { return this.operator;}
+	public FilterOperator getOperator() { return this.operator;}
 
-	public void setOperator(String operator) { this.operator = operator; }
+	public void setOperator(FilterOperator filterO) { this.operator = filterO; }
```
Comment (Hari-stackroute) [inline java/pojos/src/main/java/io/opensaber/pojos/Filter.java]: is it ok, the name filter0?

Comment (pritha-tarento) [reply java/pojos/src/main/java/io/opensaber/pojos/Filter.java]: Filter is subjected to some other entity.

File: java/pojos/src/main/java/io/opensaber/pojos/FilterOperators.java [hunk 3]
```diff
@@ -13,4 +13,8 @@
     public String getName() {
         return name;
     }
+    
+    public enum FilterOperator{
```
Comment (indrajra) [inline java/pojos/src/main/java/io/opensaber/pojos/FilterOperators.java]: what is this enum for?

File: java/registry/src/main/java/io/opensaber/registry/service/impl/SearchServiceImpl.java [hunk 4]
```diff
@@ -71,23 +72,46 @@ private SearchQuery getSearchQuery(JsonNode inputQueryNode) {
...
+
+                } else if (json.getValue().isObject()) {
+                    addToFilterList(entry.getKey(), entryVal, filterList);
+                }
+
+            }
+        }
+    }
+
+    private List<Object> getRange(Map.Entry<String, JsonNode> entry) {
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/service/impl/SearchServiceImpl.java]: I am not sure if this param needs to be of Entry type, looks a simple string is good enough.

File: java/registry/src/main/java/io/opensaber/registry/service/impl/SearchServiceImpl.java [hunk 5]
```diff
@@ -71,23 +72,46 @@ private SearchQuery getSearchQuery(JsonNode inputQueryNode) {
...
+                String operator = json.getKey();
+                Filter filter = new Filter(path);
+                String property = entry.getKey();
+                filter.setProperty(property);
+                filter.setOperator(FilterOperator.valueOf(operator));
+
+                if (json.getValue().isArray()) {
+                    List<Object> ol = getRange(json);
+                    filter.setValue(ol);
+                    filterList.add(filter);
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/service/impl/SearchServiceImpl.java]: filterList.add seems repeated within if-else. Why not keep it outside of if-else?

File: java/registry/src/main/java/io/opensaber/registry/service/impl/SearchServiceImpl.java [hunk 6]
```diff
@@ -71,23 +72,46 @@ private SearchQuery getSearchQuery(JsonNode inputQueryNode) {
...
+        Iterator<Map.Entry<String, JsonNode>> searchFields = inputQueryNode.fields();
+        // Iterate and get the fields.
+        while (searchFields.hasNext()) {
+            Map.Entry<String, JsonNode> entry = searchFields.next();
+            JsonNode entryVal = entry.getValue();
+            if (entryVal.isObject() && (entryVal.fields().hasNext())) {
+                Map.Entry<String, JsonNode> json = entryVal.fields().next();
+                String operator = json.getKey();
+                Filter filter = new Filter(path);
+                String property = entry.getKey();
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/service/impl/SearchServiceImpl.java]: String property = entry.getKey() is best placed at line 80.

Comment (pritha-tarento) [reply java/registry/src/main/java/io/opensaber/registry/service/impl/SearchServiceImpl.java]: Done

File: java/registry/src/main/java/io/opensaber/registry/service/impl/SearchServiceImpl.java (hunk 5 above)
Comment (pritha-tarento) [reply java/registry/src/main/java/io/opensaber/registry/service/impl/SearchServiceImpl.java]: Done

File: java/registry/src/main/java/io/opensaber/registry/service/impl/SearchServiceImpl.java (hunk 4 above)
Comment (pritha-tarento) [reply java/registry/src/main/java/io/opensaber/registry/service/impl/SearchServiceImpl.java]: done

File: java/pojos/src/main/java/io/opensaber/pojos/FilterOperators.java (hunk 3 above)
Comment (pritha-tarento) [reply java/pojos/src/main/java/io/opensaber/pojos/FilterOperators.java]: done

**Pull request merged**